### PR TITLE
additional clarification around trailing parameters

### PR DESCRIPTION
**Pull request by slingamn**

### CODE REVIEW

**Comment by tommyrot** [comment]
The proposal to change:

> the final parameter can be prepended by a

to:

> the final parameter MAY be prepended by a

could add confusion. I don't think emphasis is required here and the former reads better. +1 the rest.

**Comment by slingamn** [comment]
I'm open to whatever the community decides on that issue.

File: index.md [hunk 1]
```diff
@@ -346,7 +346,7 @@ This is the format of the **parameters** part, as rough ABNF:
       middle      =  nospcrlfcl *( ":" / nospcrlfcl )
       trailing    =  *( ":" / " " / nospcrlfcl )
 
-Parameters are a series of values separated by one or more ASCII SPACE characters `(' ', 0x20)`. However, to allow a value itself to contain spaces, the final parameter can be prepended by a `(':', 0x3A)` character. If the final parameter is prefixed with a colon `':'`, the prefix is stripped and the rest of the message is treated as the final parameter, no matter what characters it contains.
+Parameters are a series of values separated by one or more ASCII SPACE characters `(' ', 0x20)`. However, this syntax is insufficient in two cases: a parameter that contains one or more spaces, and an empty parameter. In order to permit such parameters, the final parameter can be prepended with a `(':', 0x3A)` character, in which case that character is stripped and the rest of the message is treated as the final parameter, including any spaces it contains. Parameters that contain spaces, are empty, or begin with a `':'` character MUST be sent as a final parameter with a preceding `':'`; in other cases, the use of a preceding `':'` for the final parameter is OPTIONAL.
```
Comment (DanielOaks) [inline index.md:349]:
```suggestion
Parameters are a series of values separated by one or more ASCII SPACE characters `(' ', 0x20)`. However, this syntax is insufficient in two cases: a parameter that contains one or more spaces, and an empty parameter. To permit such parameters, the final parameter can be prepended with a `(':', 0x3A)` character, in which case that character is stripped and the rest of the message is treated as the final parameter, including any spaces it contains. Parameters that contain spaces, are empty, or begin with a `':'` character MUST be sent with a preceding `':'`; in other cases the use of a preceding `':'` on the final parameter is OPTIONAL.
```

Minor copyedits to make the paragraph easier to read.

File: index.md [hunk 2]
```diff
@@ -362,7 +362,9 @@ Here are some examples of messages and how the parameters would be represented a
 
       :dan!d@localhost PRIVMSG #chan Hey!   ->  ["#chan", "Hey!"]
 
-As the last two examples show, a trailing parameter (a parameter prefixed with `':'`) is another regular parameter. Once the `':'` is stripped, software MUST just treat it as another param.
+      :dan!d@localhost PRIVMSG #chan ::-)   ->  ["#chan", ":-)"]
+
+As these examples show, a trailing parameter (a final parameter with a preceding `':'`) has the same semantics as any other parameter, and MUST NOT be treated specially once the `':'` is stripped.
```
Comment (DanielOaks) [inline index.md:367]:
```suggestion
      :dan!d@localhost PRIVMSG #chan ::-)   ->  ["#chan", ":-)"]

As these examples show, a trailing parameter (a final parameter with a preceding `':'`) has the same semantics as any other parameter, and MUST NOT be treated specially or stored separately once the `':'` is stripped.
```

Minor copyediting, and clarification in terms of 'stored separately' on the final bit of the explanation.

**Review by DanielOaks — COMMENTED** [review]
Minor copyedits to make this change easier to read, but nice clarifications!

**Pull request merged**